### PR TITLE
[Python Arrow] Fix issue related to TIMESTAMP_TZ and filter pushdown into PyArrow

### LINKED ISSUE #8522: [python] InvalidInputException filtering pyarrow table timestamp column with timezone and unit is not 'us'
### What happens?

When filtering a timestamp column with a timezone and the unit is not 'us' of a pyarrow table the following error appears:

```
InvalidInputException: Invalid Input Error: Attempting to execute an unsuccessful or closed pending query result

Error: Invalid Error: ArrowNotImplementedError: Function 'greater_equal' has no kernel matching input types (timestamp[ms, tz=UTC], timestamp[s])

At:
  pyarrow/error.pxi(121): pyarrow.lib.check_status
```
pyarrow version is 12.0.1

### To Reproduce

```python
import pyarrow as pa
import datetime as dt
import duckdb

t_us = pa.Table.from_arrays(
    arrays=[pa.array([dt.datetime(2022,1,1)])], 
    schema=pa.schema([pa.field("time", pa.timestamp("us", tz="UTC"))])
)

t_ms = pa.Table.from_arrays(
    arrays=[pa.array([dt.datetime(2022,1,1)])], 
    schema=pa.schema([pa.field("time", pa.timestamp("ms", tz="UTC"))])
)
```
```sql
duckdb.sql("FROM t_us").filter("time>='2022-01-01'") 

┌──────────────────────────┐
│           time           │
│ timestamp with time zone │
├──────────────────────────┤
│ 2022-01-01 00:00:00+00   │
└──────────────────────────┘
```
```sql
duckdb.sql("FROM t_ms").filter("time>='2022-01-01'") 

InvalidInputException: Invalid Input Error: Attempting to execute an unsuccessful or closed pending query result
Error: Invalid Error: ArrowNotImplementedError: Function 'greater_equal' has no kernel matching input types (timestamp[ms, tz=UTC], timestamp[s])

At:
  pyarrow/error.pxi(121): pyarrow.lib.check_status
```

### OS:

Ubuntu 22.04 x86_64

### DuckDB Version:

0.8.2-dev1435

### DuckDB Client:

python

### Full Name:

Volker Lorrmann

### Affiliation:

Siemens

### Have you tried this on the latest `master` branch?

I have tested with a master build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #8522 

Arrow supports TIMESTAMP_TZ with any time unit (s, ms, ns, us)
We only support the time unit us for them, so we convert to us.

When a filter is applied that does a comparison against a constant, the constant is cast to the type of the column.
So the constant gets cast to TIMESTAMP_TZ (us)

When this filter gets pushed down into arrow, that results in attempting to compare TIMESTAMP_TZ(ms) against TIMESTAMP_TZ(us), which is not supported.

### CODE REVIEW

**Review by hawkfish — APPROVED** [review]

**Comment by Mytherin** [comment]
Looks like the exception type is wrong in the test, could you have another look?

**Comment by Tishj** [comment]
Think I found the culprit, the test that's failing is actually underspecified, which I now fixed.
It uses `arrow_dataset().IsLoaded()` which will return true if it was loaded correctly by us before.

The test however monkeypatches the pyarrow dataset to not be present, so in the sys modules it's no longer present.
We don't check for this so the VerifyDatasetLoaded method succeeds, while later on where the attributes are imported this will fail.

I don't see why that's a huge deal though, in both cases it throws an exception, the one we throw is just a little more descriptive, none the less I changed and hopefully fixed it now :)

**Comment by Tishj** [comment]
I am tempted to remove the failing test
It's testing a scenario that we can not reliably, deterministically replicate

If the test is run standalone it passes, when I run it together with other tests, it doesn't raise an exception (because pyarrow is loaded before it was monkeypatched out)
And I'm guessing the order tests are run is platform dependent, which makes the exception we get on Linux different because of the sequence of tests that are run before the offending test.

File: tools/pythonpkg/src/python_import_cache.cpp [hunk 1]
```diff
@@ -17,7 +17,11 @@ bool PythonImportCacheItem::LoadSucceeded() const {
 
 bool PythonImportCacheItem::IsLoaded() const {
 	auto type = (*this)();
-	return type.ptr() != nullptr;
+	bool loaded = type.ptr() != nullptr;
```
Comment (hawkfish) [inline tools/pythonpkg/src/python_import_cache.cpp:20]: Not sure what the point of this change is?

**Review by hawkfish — APPROVED** [review]

File: tools/pythonpkg/src/python_import_cache.cpp (hunk 1 above)
Comment (Tishj) [reply tools/pythonpkg/src/python_import_cache.cpp:20]: None really, I had initially meant to check ModuleIsLoaded if IsLoaded returned true, to double verify
But even with monkeypatching this looked like it didn't make ModuleIsLoaded return false

So this is left ovee code that I might want to undo

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**